### PR TITLE
add abstract-eth classes

### PR DESCRIPTION
**Pull request by gianchandania**
This PR adds AbstractEthLikeMPCCoin and EthLikeMPCToken classes which will have all the common methods used by Ethlike MPC coins

### CODE REVIEW

File: modules/abstract-eth/src/ethLikeToken.ts [hunk 1]
```diff
@@ -4,14 +4,14 @@
...
-import { AbstractEthLikeCoin } from './abstractEthLikeCoin';
+import { AbstractEthLikeTssCoin } from './abstractEthLikeTssCoin';
 import { TransactionBuilder as EthTransactionBuilder, TransactionPrebuild } from '@bitgo/sdk-coin-eth';
 
 export type CoinNames = {
   [network: string]: string;
 };
 
-export class EthLikeToken extends AbstractEthLikeCoin {
+export class EthLikeToken extends AbstractEthLikeTssCoin {
```
Comment (sachushaji) [inline modules/abstract-eth/src/ethLikeToken.ts:14]: We would need an EthLikeTokenTss? will this mean all the token implementation even for multisig will have tss enabled?

File: modules/abstract-eth/src/abstractEthLikeTssCoin.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/**
+ * @prettier
+ */
+import { MPCAlgorithm } from '@bitgo/sdk-core';
+import { AbstractEthLikeCoin } from './abstractEthLikeCoin';
+
+export abstract class AbstractEthLikeTssCoin extends AbstractEthLikeCoin {
```
Comment (mohammadalfaiyazbitgo) [inline modules/abstract-eth/src/abstractEthLikeTssCoin.ts:7]:
```suggestion
export abstract class AbstractEcdsaCoin extends AbstractEthLikeCoin {
```

File: modules/abstract-eth/src/ethLikeToken.ts (hunk 1 above)
Comment (mohammadalfaiyazbitgo) [reply modules/abstract-eth/src/ethLikeToken.ts:14]: +1 on above comment

**Review by mohammadalfaiyazbitgo — COMMENTED** [review]
Using ECDSA in the class name is more descriptive than using `EthLike` imo, and do these changes need to be extended to `polygon` as well?

File: modules/abstract-eth/src/ethLikeToken.ts (hunk 1 above)
Comment (gianchandania) [reply modules/abstract-eth/src/ethLikeToken.ts:14]: added EthLikeMPCToken class

**Comment by gianchandania** [comment]
> Using ECDSA in the class name is more descriptive than using `EthLike` imo, and do these changes need to be extended to `polygon` as well?

Actually this class will be specific to EthLike coins i.e. EVM compatible coins only like arbitrum, optimism, etc. Using ECDSA in the name will give an impression that it is generic because there can be other ecdsa coins as well like cosmos, hence the name

**Review by sachushaji — APPROVED** [review]

**Pull request merged**